src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -2986,6 +2986,57 @@ public final T blockLastMillis(long timeout) {
 	}
 
 	/**
+	 * Triggers side-effects when the {@link Flux} emits an item, fails with an error or
+	 * completes successfully. This is provided as convenience to avoid using the
+	 * three corresponding {@code doOn} methods individually. Note that at least one
+	 * callback is required.
+	 *
+	 * @param nextConsumer the callback to call on {@link Subscriber#onNext(Object)}
+	 * @param errorConsumer the callback to call on {@link Subscriber#onError(Throwable)}
+	 * @param completeRunnable the callback to call on {@link Subscriber#onComplete()}
+	 * @return an observed {@link Flux}
+	 * @see #doOnNext(Consumer)
+	 * @see #doOnError(Consumer)
+	 * @see #doOnComplete(Runnable)
+	 */
+	public final Flux<T> doOnEach(Consumer<? super T> nextConsumer,
+			Consumer<? super Throwable> errorConsumer,
+			Runnable completeRunnable) {
+		if (nextConsumer == errorConsumer && errorConsumer == completeRunnable
+				&& completeRunnable == null) {
+			throw new NullPointerException("At least one side-effect callback must be non-null");
+		}
+		return doOnSignal(this, null,
+				nextConsumer, errorConsumer, completeRunnable,
+				null, null, null);
+	}
+
+	/**
+	 * Triggers side-effects when the {@link Flux} is subscribed, emits an item, fails
```

Comment:
This is really only for the advanced use cases. So if it is exposed, which I don't think is a bad idea, I would explicitly state in Javadoc that it is for advanced used cases and that generally the other more specific `doOnXX` should be used instead, else we risk people using this by accident and over-complicating things.